experiments/multichain/database_reader.py

```diff
@@ -0,0 +1,130 @@
+import os
+from os import path
+import base64
+
+from Tribler.community.multichain.database import MultiChainDB
+
+
+class MultiChainExperimentAnalysisDatabase(MultiChainDB):
+    """
+    Extended MultiChainDB that provides additional functionality to analyze an experiment.
+    """
+
+    def __init__(self, dispersy, working_directory):
+        super(MultiChainExperimentAnalysisDatabase, self).__init__(dispersy, working_directory)
+
+    def get_all_blocks(self):
+        """
+        Returns all blocks form the database
+        """
+        db_query = u"SELECT public_key_requester, public_key_responder, up, down, " \
+                   u"total_up_requester, total_down_requester, sequence_number_requester, previous_hash_requester, " \
+                   u"signature_requester, hash_requester, " \
+                   u"total_up_responder, total_down_responder, sequence_number_responder, previous_hash_responder, " \
+                   u"signature_responder, hash_responder, insert_time " \
+                   u"FROM `multi_chain`"
+        db_result = self.execute(db_query).fetchall()
+        return [self._create_database_block(result) for result in db_result]
+
+
+class DatabaseReader(object):
+
+    def __init__(self, working_directory):
+        """
+        Parent class that contains functionality for both gumby and single database readers.
+        DatabaseReaders process experimentation information.
+        """
+        # Either contains all information all ready or all information will be aggregated here.
+        self.working_directory = working_directory
+        self.database = self.get_database(self.working_directory)
+
+        self.combine_databases()
```

Comment:
Maybe you could move these 2 calls outside of the `__init__` and call them after the DatabaseReader is constructed. This way the `__init__` doesn't do any processing.